src/topology/sheaves/sheaf_condition/sites.lean

```diff
@@ -472,4 +473,55 @@ begin
       nat_trans.id_app G.1, category.comp_id, nat_trans.id_app F.1, category.id_comp], },
 end
 
+
 end Top.presheaf
+
+
+namespace Top.opens
+
+open category_theory topological_space
+
+variables {X : Top} {ι : Type*}
+
+lemma cover_dense_iff_basis [category ι] (B : ι ⥤ opens X) :
+  cover_dense (opens.grothendieck_topology X) B ↔ opens.is_basis (set.range B.obj) :=
+begin
+  rw opens.is_basis_iff_nbhd,
+  split, intros hd U x hx, rcases hd.1 U x hx with ⟨V,f,⟨i,f₁,f₂,hc⟩,hV⟩,
+  exact ⟨B.obj i, ⟨i,rfl⟩, f₁.le hV, f₂.le⟩,
+  intro hb, split, intros U x hx, rcases hb hx with ⟨_,⟨i,rfl⟩,hx,hi⟩,
+  exact ⟨B.obj i, ⟨⟨hi⟩⟩, ⟨⟨i, 𝟙 _, ⟨⟨hi⟩⟩, rfl⟩⟩, hx⟩,
+end
+
+lemma cover_dense_induced_functor {B : ι → opens X} (h : opens.is_basis (set.range B)) :
+  cover_dense (opens.grothendieck_topology X) (induced_functor B) :=
+(cover_dense_iff_basis _).2 h
+
+end Top.opens
+
+namespace Top.sheaf
+
+open category_theory topological_space Top opposite
+
+variables {C : Type u} [category.{v} C] [limits.has_products C]
+variables {X : Top.{v}} {ι : Type*} {B : ι → opens X}
+variables (F : presheaf C X) (F' : sheaf C X) (h : opens.is_basis (set.range B))
+
+/-- If a family `B` of open sets forms a basis of the topology on `X`, and if `F'`
+    is a sheaf on `X`, then a homomorphism between a presheaf `F` on `X` and `F'`
+    is equivalent to a homomorphism between their restrictions to the indexing type
+    `ι` of `B`, with the induced category structure on `ι`. -/
+def restrict_hom_equiv_hom :
+  ((induced_functor B).op ⋙ F ⟶ (induced_functor B).op ⋙ F'.1) ≃ (F ⟶ F'.1) :=
+@cover_dense.restrict_hom_equiv_hom _ _ _ _ _ _ _ _ (opens.cover_dense_induced_functor h)
```

Comment:
This `@` suggests that maybe `cover_dense.restrict_hom_equiv_hom` needs more explicit arguments?

Reply:
The sites version of `restrict_hom_equiv_hom` has `F` and `F'` implicit, and here only the `F'` argument (namely `(presheaf.Sheaf_spaces_to_sheaf_sites C X).obj F'`) can't be inferred. I think the argument is gonna be made explicit only at this point where we bridge sites and spaces.

However, the spaces version I added has both `F` and `F'` explicit; this is because in [my Gamma-Spec work](https://github.com/leanprover-community/mathlib/pull/9802/files#diff-8acd5cf78fed0a9c13227af5f70437a7b0dac964a4d6f3186792ac3b19a5ae95R138-R139) I found that supplying one of them helped Lean to infer `B` `basic_open` (and `F` has to come before `h` to make this work, which is also inconsistent with the sites version). If I don't supply `(structure_sheaf (Γ' X)).1`, the following error would appear:
```
type mismatch at application
  Top.sheaf.restrict_hom_equiv_hom ?m_7 ?m_8 is_basis_basic_opens
term
  is_basis_basic_opens
has type
  opens.is_basis (set.range basic_open)
but is expected to have type
  opens.is_basis (set.range ?m_3)
Additional information:
.../mathlib/src/algebraic_geometry/Gamma_Spec_adjunction.lean:138:7: context: switched to simple application elaboration procedure because failed to use expected type to elaborate it, error message
  too many arguments
 ```
Same error occurs even if I supply implicit argument `Γ' X` to `is_basis_basic_opens` does nothing, or change `λ r : R, basic_open r` to `@basic_open R _` in the definition of `is_basis_basic_opens`.

Overall, if consistency (of argument explicitness/order) isn't a strict requirement, I think the current status is satisfactory.

Reply:
Ok, thanks for the explanation!